### PR TITLE
Fixed the redirection behavior

### PR DESCRIPTION
**Pull request by czardoz**
## PR Checklist:
- [x] I have run `npm test` locally and all tests are passing.
- [x] I have added/updated tests for any new behavior.
       
- [x] If this is a significant change, an issue has already been created where the problem / solution was discussed: [N/A, or add link to issue here]

### CODE REVIEW

**Comment by czardoz** [comment]
Fixes https://github.com/postmanlabs/postman-request/pull/12/files

File: lib/redirect.js [hunk 1]
```diff
@@ -112,6 +112,13 @@ Redirect.prototype.onResponse = function (response) {
     delete request.agent
   }
 
+  // remove the host header if the response is a 307 & the host has changed.
+  // (the new host will be populated auto-magically).
+  // For other types of redirects, it's removed automatically (see code below)
+  if (response.statusCode === 307 && request.headers && uriPrev.host !== request.uri.host) {
```
Comment (numaanashraf) [inline lib/redirect.js]: Wouldn't this same issue be there for 301, 302 & 308 too?

File: tests/test-follow-307.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+var port
+
+tape('setup', function (t) {
+  server.listen(0, function () {
+    port = this.address().port
+    t.end()
+  })
+})
+
+tape('307 redirect should work when host is set explicitly, but changes on redirect', function (t) {
```
Comment (numaanashraf) [inline tests/test-follow-307.js]: Can we add similar cases for 301, 302 & 308 too?

File: lib/redirect.js (hunk 1 above)
Comment (czardoz) [inline lib/redirect.js]: It's already handled (check comment above)

File: tests/test-follow-307.js (hunk 2 above)
Comment (czardoz) [inline tests/test-follow-307.js]: Already present, no special handling is needed for those

**Comment by czardoz** [comment]
Also, fixes https://github.com/postmanlabs/postman-app-support/issues/2550

**Pull request merged**